### ISSUE TITLE
Unable to bind to GREE Bora A5 - GWH09AAB-K6DNA5A/I

### ISSUE THREAD

**Issue by KiralyCraft** [cmroche/greeclimate#70]
First and foremost, thank you for this beautiful piece of software. The integration with HA is absolutely marvelous. This integration is the sole reason for which I bought a GREE device, moving from a portable unit to a wall-mounted one. 

Unfortunately, it seems that HA doesn't want to bind to the device, which seems to be running firmware version V1.21; I've tested the standalone variant, using "gree.py" and it seems it fails to bind. 

Here's the output of that:
(I added a print before attempting to bind, to confirm I was doing it right)

```
asyncio - DEBUG - Using selector: EpollSelector
__main__ - DEBUG - Scanning network for Gree devices
greeclimate.discovery - INFO - Scanning for Gree devices ...
greeclimate.discovery - DEBUG - Listening for devices on 192.168.0.255
greeclimate.network - DEBUG - Sending packet:
{"t": "scan"}
greeclimate.network - DEBUG - Received packet from 192.168.0.131:
{"t": "pack", "i": 1, "uid": 0, "cid": "", "tcid": "", "pack": {"t": "dev", "cid": "9424b8b6b00f", "bc": "00000000000000000000000000000000", "brand": "gree", "catalog": "gree", "mac": "9424b8b6b00f", "mid": "10001", "model": "gree", "name": "", "lock": 0, "series": "gree", "vender": "1", "ver": "V2.0.0", "ModelType": "32772", "hid": "362001065279+U-WB05RT13V1.21.bin"}}
greeclimate.discovery - INFO - Found gree device Device: 9424b8b6b00f @ 192.168.0.131:7000 (mac: 9424b8b6b00f)
__main__ - INFO - Attempting to bind NOW!
greeclimate.device - INFO - Starting device binding to Device: 9424b8b6b00f @ 192.168.0.131:7000 (mac: 9424b8b6b00f)
greeclimate.network - DEBUG - Sending packet:
{"cid": "app", "i": 1, "t": "pack", "uid": 0, "tcid": "9424b8b6b00f", "pack": {"mac": "9424b8b6b00f", "t": "bind", "uid": 0}}
greeclimate.network - DEBUG - Received packet from 192.168.0.245:
{"t": "pack", "i": 1, "uid": 0, "cid": "", "tcid": "", "pack": {"t": "dev", "cid": "9424b8b6b21f", "bc": "00000000000000000000000000000000", "brand": "gree", "catalog": "gree", "mac": "9424b8b6b21f", "mid": "10001", "model": "gree", "name": "", "lock": 0, "series": "gree", "vender": "1", "ver": "V2.0.0", "ModelType": "32772", "hid": "362001065279+U-WB05RT13V1.21.bin"}}
greeclimate.discovery - INFO - Found gree device Device: 9424b8b6b21f @ 192.168.0.245:7000 (mac: 9424b8b6b21f)
__main__ - INFO - Attempting to bind NOW!
greeclimate.device - INFO - Starting device binding to Device: 9424b8b6b21f @ 192.168.0.245:7000 (mac: 9424b8b6b21f)
greeclimate.network - DEBUG - Sending packet:
{"cid": "app", "i": 1, "t": "pack", "uid": 0, "tcid": "9424b8b6b21f", "pack": {"mac": "9424b8b6b21f", "t": "bind", "uid": 0}}
greeclimate.network - DEBUG - Received packet from 192.168.0.248:
{"t": "pack", "i": 1, "uid": 0, "cid": "", "tcid": "", "pack": {"t": "dev", "cid": "9424b8b6b1e9", "bc": "00000000000000000000000000000000", "brand": "gree", "catalog": "gree", "mac": "9424b8b6b1e9", "mid": "10001", "model": "gree", "name": "", "lock": 0, "series": "gree", "vender": "1", "ver": "V2.0.0", "ModelType": "32772", "hid": "362001065279+U-WB05RT13V1.21.bin"}}
greeclimate.discovery - INFO - Found gree device Device: 9424b8b6b1e9 @ 192.168.0.248:7000 (mac: 9424b8b6b1e9)
__main__ - INFO - Attempting to bind NOW!
greeclimate.device - INFO - Starting device binding to Device: 9424b8b6b1e9 @ 192.168.0.248:7000 (mac: 9424b8b6b1e9)
greeclimate.network - DEBUG - Sending packet:
{"cid": "app", "i": 1, "t": "pack", "uid": 0, "tcid": "9424b8b6b1e9", "pack": {"mac": "9424b8b6b1e9", "t": "bind", "uid": 0}}
__main__ - INFO - Done discovering devices
```

I never get a response regarding the firmware version. It works fine with the EWPE Smart application. Due to the delay, however, it makes me believe it's communicating over the cloud, and not locally. 

The details of my Wifi adapter are:

- Model: GRJWB04-J
- Not sure: WB05RT13V1.21

**Comment by KiralyCraft** [comment]
I've given this fork of another project a shot, it seems the problem is not related to this, the device does not respond to this attempt either: https://github.com/stas-demydiuk/ewpe-smart-mqtt/pull/34

I managed to extract the `gree.db` database from EWPE Smart 1.18.6.5, trying to see if using the encryption key already established would return something. Unfortunately, it seems this database is encrypted now, unlike previous versions.

**Comment by KiralyCraft** [comment]
This issue may be related, the device OP is using seems similar to mine. I'll inquiry about their firmware version.

https://github.com/home-assistant/core/issues/109230

**Comment by KiralyCraft** [comment]
This seems to be fixed with test version 1.53. There may have been a bug in firmware 1.2X; This version is not yet available as a release, but rather only for applications put in debug mode. Therefore, my specific issue is now solved, since I've updated my ACs to version 1.53, and they now work without further modification.
